mempool/src/pool/tests.rs

```diff
@@ -0,0 +1,1787 @@
+use super::*;
+use common::chain::signature::inputsig::InputWitness;
+use common::chain::transaction::{Destination, TxInput, TxOutput};
+use common::chain::OutPointSourceId;
+use common::chain::OutputPurpose;
+use core::panic;
+use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::Arc;
+
+const DUMMY_WITNESS_MSG: &[u8] = b"dummy_witness_msg";
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+struct ValuedOutPoint {
```

Comment:
Can you move all this "machinery" to a utils submodule, and just keep the tests in this file?

All in the interest of making this file smaller... it's like 1800 lines... we should split this down to multiple parts each with its subcategory of tests.

Reply:
Done.